src/OgGroupAudienceHelper.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\og\OgGroupAudienceHelper.
+ */
+
+namespace Drupal\og;
+
+use Drupal\Core\Entity\ContentEntityInterface;
+use Drupal\Core\Field\FieldException;
+use Drupal\Core\Field\FieldStorageDefinitionInterface;
+
+/**
+ * OG audience helper methods.
+ */
+class OgGroupAudienceHelper {
+
+  /**
+   * Return TRUE if a field can be used and has not reached maximum values.
+   *
+   * @param \Drupal\Core\Entity\ContentEntityInterface $entity
+   *   The content entity to check the field cardinality for.
+   * @param string $field_name
+   *   The field name to check the cardinality of.
+   *
+   * @return bool
+   *
+   * @throws \Drupal\Core\Field\FieldException
+   */
+  public static function checkFieldCardinality(ContentEntityInterface $entity, $field_name) {
+    $field_definition = $entity->getFieldDefinition($field_name);
+
+    if (!$field_definition) {
+      throw new FieldException(sprintf('No "%s" field found for %s %s entity', $field_name, $entity->bundle(), $entity->getEntityTypeId()));
```

Comment:
I see your point, but in terms of readability it easier to follow a [sentence](https://github.com/amitaibu/og/blob/8.x-1.x/src/Og.php#L408) rather than mentally replacing the `%s` with an argument.